### PR TITLE
feat(ci): Add e2e test bed

### PR DESCRIPTION
**Pull request by jamilbk**
- [x] Launch control plane via docker compose
- [x] Ensure all clients build

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -1,85 +1,209 @@
...
+          cp Firezone/xcconfig/debug.xcconfig Firezone/xcconfig/config.xcconfig
 
-  windows:
-    needs: setup-e2e
-    runs-on: windows-11-firezone
-    steps:
-      - uses: actions/checkout@v4
-      - run: echo 'Noop'
+          # Unlock keychain; simulates a user logged into the GUI
+          security unlock-keychain -p '${{ secrets.MACOS_SELF_HOSTED_USER_PASSWORD }}'
```
Comment (ReactorScram) [inline .github/workflows/e2e.yml:68]: "MACOS"?

**Review by ReactorScram — APPROVED** [review]
The Windows stuff looks good. So this just builds a debug package, and the next big change would be to install the MSI and smoke-test it?

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (jamilbk) [reply .github/workflows/e2e.yml:68]: Yeah this is the local user account password for the `firezone` user as stored in GH secrets.

**Comment by jamilbk** [comment]
> So this just builds a debug package, and the next big change would be to install the MSI and smoke-test it?

Correct. Though we may not even need to do that. See https://tauri.app/v1/guides/testing/webdriver/introduction

**Pull request merged**